server/routes/mentions.js

```diff
@@ -0,0 +1,76 @@
+const router = require("express").Router();
+
+const requiresAuth = require("./middleware/requiresAuth");
+const { Mention, User } = require("../models");
+
+const callScraper = require("../scraper");
+
+// TODO test mentions in action
+
+router.get("/", requiresAuth, async (req, res) => {
+  const user = await User.findByPk(req.user.id);
+  // note that the getter method for n-m can accept options
+  // as if it were a regular query
+  const mentions = await user.getMentions();
+
+  res.json(mentions);
+});
+
+router.get("/:id", requiresAuth, async (req, res) => {
+  const mention = await Mention.findByPk(req.params.id);
+
+  res.json(mention);
+});
+
+router.post("/", requiresAuth, async (req, res) => {
+  // NOTE I think this is better served as just a test route in order to test the scraper
+  // This shouldn't be placed in the final PR as there will be another means to post new data
+  // into the database.
+  const user = await User.findByPk(req.user.id);
+
+  let mentions = callScraper(req.user.company, 100);
+
+  let result = [];
+  for (let m of mentions) {
+    let userMentions = await user.getMentions({
+      where: {
+        title: m.title,
+      },
+    });
+
+    // if this is new
+    if (!userMentions.length) {
+      [mention, isNew] = await Mention.findOrCreate({
+        where: {
+          title: m.title,
+        },
+        defaults: {
+          title: m.title,
+          platform: m.platform,
+          date: m.date,
+          content: m.content || "",
+          popularity: m.popularity,
+          imageUrl: m.image,
+        },
+      });
+      result.push(mention);
+      user.addMention(mention);
+    }
+  }
+  res.json({ mentions: result });
+});
+
+router.delete("/:id", requiresAuth, async (req, res) => {
+  const user = await User.findByPk(req.user.id);
+
+  // found a way to untag a given mention from user,
+  // while preserving the mention on its own separate table
```

Comment:
Since this is the router for mentions, I guess this delete route is supposed to delete a mention from DB instead of untagging it from the logged in user. And for the latter, we may not need such a functionality for untagging a certain mention from a certain user. 
I don't know, there may be some irrelevant mentions such as a mention about the item in WOW while the user's input is about the card game. In this case, it would be better if we could untag it from the user, but we will have to do it manually, since if there exsits any automatic approach to achieve that, we would have applied it in the scraper and the irrelevent mentions wouldn't have been added into DB at the first place. But is doing manually practical and is want we want? Let me know what you think.

Reply:
Since Users and Mentions have a M-N relationship, in situations where a Mention can have many users' companies we might not want to delete the mention completely where while one user might no longer require it, another different user still might. 

I was thinking that actually deleting mentions (deleting the table entry instead of untagging them from a user in the UserMentions through table, which is useful in instances such as when mentions are considered stale) is better served on a separate background worker responsible for checking in on the database and removing entries no longer needed. 

What I had in mind for this route is providing the user a means to remove mentions that are not relevant to them, while still preserving said mention for another user that might still use it.

Reply:
I got what you meant. We could have 2 routes for deleting mentions, one for actually deleting a mention, the other for untagging it from a user. But the latter may not be required by our app at this moment. For now, the mockup doesn't provide any functionality for user to untag a certian mention. And I don't see it practical in the future. For example, if the mentions displayed in the main page are from the DB, and a user deleted one of them, then what would happen in the next auto-scraping? The just deleted mention would be obtained and added into DB again. Unless we keep a list for all the mentions deleted by the user, the deletion operation is not very practical.

Reply:
> But the latter may not be required by our app at this moment. For now, the mockup doesn't provide any functionality for user to untag a certian mention. And I don't see it practical in the future.

Fair enough, it is not specified in the ticket requirement. I will remove it from the PR